Add QNetworkConfigurationManager-based connection detection

hagabaka opened the pull request:
A new network detection mode is added to use QNetworkConfigurationManager, available in Qt4.7+ including Qt5. This becomes the default mode when KDE4 integration is not available.

CMakeLists.txt just assumes the availability of QNetworkConfigurationManager based on Qt version. I haven't figured out how to actually check it, and I don't know if it will be necessary.

I've tested that it compiles with -DUSE_QT4 or -DUSE_QT5.

By the way, Networking is not available in KF5 Solid, and [porting notes](https://community.kde.org/Frameworks/Porting_Notes#Solid_Networking) point to QNetworkConfigurationManager.

hagabaka commented:
It turns out I wasn't actually testing Qt4 when I created the pull request due to CMake caching. But after the above fixes, it does build for both Qt4 and Qt5 now, and automatic disconnection/reconnection work as expected.

Sput42 commented:
Great stuff! Also good to know that it's the preferred way of doing it also for KF5.

However, I think we can make things simpler now: since we now rely on Qt 4.8+, there is no need to check for existence of the class. So the checks and the #define could be removed, making things much more readable.

I'd appreciate if you could do that and maybe afterwards squash the commits together :) Thanks!

hagabaka commented:
I've removed the #defines and squashed the commits. The build is failing but that seems to be because of quassel/quassel@ef13687df76b292f36f330bed44cf9f35ac53632

Sput42 commented:
Thanks a lot!
